### ISSUE TITLE
Using Scikit-Learn on Tensorflow 2.0-GPU with Neuraxle

### ISSUE THREAD

**Issue by PythonCommander** [freqtrade/freqtrade#3704]
## Describe your environment
(if applicable)

  * Operating system: ____
  * Python Version: _____ (`python -V`)
  * CCXT version: _____ (`pip freeze | grep ccxt`)
  * Freqtrade Version: ____ (`freqtrade -V` or `docker-compose run --rm freqtrade -V` for Freqtrade running in docker)

## Describe the enhancement

Hello, I was trying to find a way to run hyperopt on a gpu to optimize a strategy faster. I recently ran into this article saying that you can access scikit-learn library on Tensorflow 2.0 by installing the Neuraxle extension. If this is possible, I thought I'd recommend the person's response to the Freqtrade team.

"Yes, if you use TensorFlow or PyTorch within Scikit-Learn pipelines by using [Neuraxle](https://github.com/Neuraxio/Neuraxle).
Neuraxle is an extension of Scikit-Learn to make it more compatible with all deep learning libraries.

Here is a full project example from A to Z where [TensorFlow is used with Neuraxle as if it was used with Scikit-Learn.](https://github.com/guillaume-chevalier/seq2seq-signal-prediction)

Here is another practical example where TensorFlow is used within a scikit-learn-like pipeline

The trick is performed by using [Neuraxle-TensorFlow](https://github.com/Neuraxio/Neuraxle-TensorFlow) or [Neuraxle-PyTorch](https://github.com/Neuraxio/Neuraxle-PyTorch).

Why so?
Using one of Neuraxle-TensorFlow or Neuraxle-PyTorch will provide you with a saver to allow your thing to be serialized correctly. You want it to be serialized correctly to be able to ensure compatibility between scikit-learn and your Deep Learning framework when it comes time to save or parallelize things and so forth. You can read how Neuraxle solves this with savers [here](https://www.neuraxle.org/stable/scikit-learn_problems_solutions.html#problem-you-can-t-parallelize-nor-save-pipelines-using-steps-that-can-t-be-serialized-as-is-by-joblib)."

**Comment by xmatthias** [comment]
so i assume this did not work out in the end?

How far did you get? 
what where the problems that "broke" this iniciative?

**Comment by PythonCommander** [comment]
My programming skills are so limited right now. I only have 5 months into writing code. I looked through hyperopt and backtesting files and I realized that the gpu requires numerical datasets and test sets to do things like regression and reinforcement learning. The freqtrade code was more procedural so I would have to change the code to everything entirely to even get the gpu to accept and  backtest the data. At this point I'm really not sure though I can be completely wrong. Maybe one day when I get better at programming, I can contribute and start a regression code for freqtrade gpu.
